### PR TITLE
L226 en distribution usa

### PR DESCRIPTION
**Pull request by abigailsnyder**
#721 

needs careful review; some tibbles that I think should be included as outputs were not included in the skeleton So they've been created but not output (eg lines 248-265). The inputs that are tagged as not appearing as precursors are related to these.
Not to mention I don't know what any of the outputs do in gcam, so my comments are basically useless. 

additionally, a few of my changes trigger warnings that I'm not sure how to surpress:
I adjusted write_to_all_states so that it doesn't /have/ to remove the region column if it's not present; this causes a warning but works fine (the calls on lines 287 and 296).
My call to separate on line 219 works fine, but probably should have some additional lines first to smoosh some data and then after to separate the smooshed data back out, to avoid the warnings that come up from it

### CODE REVIEW

**Comment by abigailsnyder** [comment]
@bpbond I'm going to work on cleaning up some of these warnings when I do my DSR hours on Friday; then I'll work on cleaning the AgLU code.

**Comment by bpbond** [comment]
Thanks, Abigail.

File: R/constants.R [hunk 1]
```diff
@@ -216,7 +216,7 @@ aglu.MGMT_LOGIT_TYPE <- "absolute-cost-logit"
 # XML-related constants
 aglu.GLU_NDIGITS          <- 3    # number of digits in the geographic land unit identifier codes
 aglu.LT_GLU_DELIMITER     <-      # delimiter between the land use type name and GLU name. should be the same as the crop-glu delimiter
-aglu.CROP_GLU_DELIMITER   <- "_"  # delimiter between the crop name and GLU name
+  aglu.CROP_GLU_DELIMITER   <- "_"  # delimiter between the crop name and GLU name
```
Comment (gokuliyer) [inline R/constants.R]: Did something really change here?

File: R/zchunk_L226.en_distribution_USA.R [hunk 2]
```diff
@@ -1,20 +1,35 @@
 #' module_gcam.usa_L226.en_distribution_USA
 #'
-#' Briefly describe what this chunk does.
+#' This chunk creates L226.DeleteSupplysector_USAelec, L226.StubTechCoef_electd_USA, L226.TechShrwt_electd_USA, L226.TechCost_electd_USA,
```
Comment (gokuliyer) [inline R/zchunk_L226.en_distribution_USA.R]: Can we make this a little more descriptive and summarize information from below?

File: R/zchunk_L226.en_distribution_USA.R [hunk 3]
```diff
@@ -63,165 +78,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
...
+    # L226.TechCoef_en_USA: technology coefficients and market names of energy handling and delivery
+    L226.TechShrwt_en_USA %>%
+      select(one_of(LEVEL2_DATA_NAMES[["TechYr"]])) %>%
+      mutate(minicam.energy.input = supplysector,
+             coefficient = gcamusa.DEFAULT_COEFFICIENT,
+             market.name = gcamusa.DEFAULT_MARKET) ->
+      L226.TechCoef_en_USA
+
+
+    # L226.CostAdj_75USDGJ_FERC_F: grid region specific cost adders
```
Comment (gokuliyer) [inline R/zchunk_L226.en_distribution_USA.R]: A quick description of how the cost adders are calculated (based on calculation below) would help here.

**Comment by gokuliyer** [comment]
@bpbond : Abigail mentions in an earlier comment that this needs careful review. From a disciplinary perspective, the chunk seems fine, except for requests to do add additional comments. I assume that you took care of the coding side of the chunk? I did not see anything incorrect; but if there's a need for my input on something, please let me know.

**Comment by abigailsnyder** [comment]
@bpbond @gokuliyer are the tables 
`L226.Supplysector_electd_USA`
`L226.SubsectorLogit_electd_USA` 
`L226.SubsectorShrwtFllt_electd_USA`
`L226.SubsectorInterp_electd_USA` 
actually needed outputs of this chunk that the skeleton just missed?

File: R/zchunk_L226.en_distribution_USA.R (hunk 3 above)
Comment (abigailsnyder) [reply R/zchunk_L226.en_distribution_USA.R]: The comments detailing this are already present, but I've slightly updated to make it clearer that that is what they are

File: R/zchunk_L226.en_distribution_USA.R [hunk 4]
```diff
@@ -63,165 +79,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
...
+
+
+    # global_energy_to_USA_electd - takes global energy inputs from L226.en_distribution.R
+    # and processes for use in USA electricity T&D
+    global_energy_to_USA_electd <- function(data){
+      data %>%
+        filter(region == "USA",
+               supplysector %in% gcamusa.ELECT_TD_SECTORS) %>%
+        write_to_all_states(names(data)) ->
+        datanew
```
Comment (bpbond) [inline R/zchunk_L226.en_distribution_USA.R]: Just end function with `write_to_all_states` call, no need to assign to `datanew`

File: R/zchunk_L226.en_distribution_USA.R [hunk 5]
```diff
@@ -63,165 +79,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
...
+
+    # Supplysector information
+
+    # PART 1: FUEL HANDLING AND DELIVERY SECTORS
+
+    # L226.Supplysector_en_USA: Supply sector information for energy handling and delivery sectors
+    # NOTE: Currently using FERC regions as a proxy for regional energy markets
+    A21.sector %>%
+      select(supplysector, output.unit, input.unit, price.unit, logit.exponent, logit.type) %>%
+      filter(supplysector %in%gcamusa.REGIONAL_FUEL_MARKETS) ->
```
Comment (bpbond) [inline R/zchunk_L226.en_distribution_USA.R]: ` %in% `

**Comment by abigailsnyder** [comment]
uh oh @bpbond things got borked for me since we were both pushing to this. I have conflicts I can't figure out how to resolve to push the rest of my changes

File: R/zchunk_L226.en_distribution_USA.R [hunk 6]
```diff
@@ -63,165 +79,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
...
+      A21.tmp
+
+    A26.sector  %>%
+      select(supplysector, output.unit, input.unit, price.unit, logit.exponent, logit.type) %>%
+      filter(supplysector %in%gcamusa.REGIONAL_FUEL_MARKETS) %>%
```
Comment (bpbond) [inline R/zchunk_L226.en_distribution_USA.R]: As above - space after `%in%` pls

File: R/zchunk_L226.en_distribution_USA.R [hunk 7]
```diff
@@ -63,165 +79,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
...
+      bind_rows(A21.tmp, .) %>%
```
Comment (bpbond) [inline R/zchunk_L226.en_distribution_USA.R]: ? Does binding order matter?

**Comment by bpbond** [comment]
Ah crap my bad. I'll come by in a minute/

File: R/zchunk_L226.en_distribution_USA.R [hunk 8]
```diff
@@ -63,165 +79,411 @@ module_gcam.usa_L226.en_distribution_USA_DISABLED <- function(command, ...) {
...
+
+    # L226.DeleteSupplysector_USAelec: Removing the electricity T&D sectors of the USA region
+    # This should probably be converted to an assumption and read in at some point.
+    L226.DeleteSupplysector_USAelec <- tibble(region = "USA", supplysector = gcamusa.ELECT_TD_SECTORS)
+
+
+    ### Replacing for loop starting on line 152 in old DS. I /think/ these tibbles should be output in place of the "object" that currently is;
+    ### Otherwise just delete all of the below to line 250
+    ### There's also a couple inputs to this chunk that are NULL, and nothing gets done to: L226.SubsectorShrwt_en, L226.SubsectorInterpTo_en
+    L226.Supplysector_en %>%
```
Comment (bpbond) [inline R/zchunk_L226.en_distribution_USA.R]: Your call but I'd just do these in one-liners, e.g. `L226.Supplysector_electd_USA <- global_energy_to_USA_electd(L226.Supplysector_en)`

**Comment by bpbond** [comment]
Doh. 👏 not 🗜 obviously

**Comment by abigailsnyder** [comment]
now I just wish that the emoji for clamps is the robot from futurama

**Comment by abigailsnyder** [comment]
Ok @bpbond all comments addressed and now passing locally on my machine

**Comment by abigailsnyder** [comment]
@bpbond I have no idea why this is failing on travis; you may have to check it out and test it on your machine to see if it's a windows issue. It's passing on my machine still

**Comment by gokuliyer** [comment]
@bpbond : Yes.

**Comment by abigailsnyder** [comment]
@bpbond was it really just the GCAM_DATA_MAP update needed to pass Travis? I forgot about that whole kerfuffle, I'll keep it in mind for my next chunk

**Comment by abigailsnyder** [comment]
oh heck now outputs are failing, shall I checkout and track them down?

**Comment by bpbond** [comment]
Hopefully! It passes everything on my machine. 👍

**Comment by bpbond** [comment]
...and no. Hmm.

**Comment by bpbond** [comment]
Let me look first. Thanks.

**Comment by bpbond** [comment]
Yes, two outputs failing:

```
Failed -------------------------------------------------------------------------
1. Failure: matches old data system output (@test_oldnew.R#132) ----------------
dim(olddata) not identical to dim(newdata).
1/2 mismatches
[1] 3367 - 3366 == 1
Dimensions are not the same for L226.TechCoef_electd_USA.csv

2. Failure: matches old data system output (@test_oldnew.R#145) ----------------
round_df(olddata) not equivalent to round_df(newdata).
Incompatible type for column `year`: x character, y numeric
Incompatible type for column `coefficient`: x character, y numeric
L226.TechCoef_electd_USA.csv doesn't match
```

These both _look_ pretty easy to resolve, on the face of it, but I need to get on a 2 PM call. If you have time to look @abigailsnyder great; otherwise I will later. Thanks.

**Comment by abigailsnyder** [comment]
I'll take a look @bpbond

**Comment by abigailsnyder** [comment]
@bpbond VERY easy - I screwed up bringing the comparison data over and repeated the row of column names. I'm double checking on my machine and then I'll push

**Comment by bpbond** [comment]
🎉

**Pull request merged**